app/models/grda_warehouse/system_cohorts/currently_homeless.rb

```diff
@@ -0,0 +1,125 @@
+        ongoing.
+        where.not(client_id: service_history_source.where(date: Date.yesterday, homeless: false).select(:client_id)).
+        where.not(client_id: moved_in_ph).
+        where.not(client_id: cohort_clients.select(:client_id)).
+        group(:client_id).minimum(:first_date_in_program)
+
+      previous_enrollments = homeless_enrollment_source.
+        exit.
+        where(client_id: candidate_enrollments.keys).
+        order(greatest(she_t[:last_date_in_program], she_t[:move_in_date])).
+        pluck(:client_id, greatest(she_t[:last_date_in_program], she_t[:move_in_date]), :destination).
+        map { |client_id, *rest| [client_id, rest] }.to_h
+
+      most_recent_service_dates = service_history_source.
+        where(client_id: candidate_enrollments.keys).
+        group(:client_id).maximum(:date)
+
+      newly_identified = []
+      returned_from_housing = []
+      returned_from_inactive = []
+
+      candidate_enrollments.each do |client_id, enrollment_date|
+        previous_service_date, previous_destination = previous_enrollments[client_id]
+        if previous_service_date.blank? || previous_service_date < enrollment_date - 2.years
+          newly_identified << client_id
+        elsif HUD.permanent_destinations.include?(previous_destination) || previous_service_date < enrollment_date
+          returned_from_housing << client_id
+        elsif most_recent_service_dates[client_id] >= Date.current - days_of_inactivity.days
+          returned_from_inactive << client_id
+        end
+      end
+
+      add_clients(newly_identified, 'Newly identified')
+      add_clients(returned_from_housing, 'Returned from housing')
+      add_clients(returned_from_inactive, 'Returned from inactive')
+    end
+
+    private def remove_housed_clients
+      # Housed (received a move-in date in a PH project, or exited to a Permanent destination from one of their homeless projects).
+      # Limit to enrollments started after date added to cohort
+      cohort_enrollments = enrollment_source.where(
+        client_id: cohort_clients.joins(client: :service_history_enrollments).
+          where(she_t[:first_date_in_program].gt(c_client_t[:date_added_to_cohort])).
+          select(:client_id),
+      )
+
+      moved_in = cohort_enrollments.ph.where.not(move_in_date: nil).pluck(:client_id)
+      with_permanent_destination = cohort_enrollments.homeless.exit.where(destination: HUD.permanent_destinations).pluck(:client_id)
+      remove_clients(moved_in | with_permanent_destination, 'Housed')
+    end
+
+    private def remove_inactive_clients
+      # Inactive (hasn't been seen in a homeless project in N days, where N refers to the setting on the cohort.)
+      inactive_date = Date.current - days_of_inactivity.days
+      client_ids = enrollment_source.
+        homeless.
+        where(client_id: cohort_clients.select(:client_id)).
+        joins(:service_history_services).
+        where(shs_t[:date].gt(inactive_date)).
```

Comment:
again I _think_ `shs_t[:date].lt(inactive_date)`

Reply:
Maybe I had the logic backwards, this is finding someone who has service after the day they became inactive, thus making them actually active?

Reply:
Yes!